Packages/Core/Editor/Drawers/UsageIndex.cs

```diff
@@ -0,0 +1,22 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace UnityAtoms.Editor
+{
+	public partial class AtomBaseReferenceDrawer
+    {
+        private class UsageIndex
+        {
+            public int Index { get => GetIndex(); set => SetIndex(value); }
```

Comment:
Fair point, I just didn't like doing it like so at the time, don't remember why exactly, maybe it's because it wasn't just about the index but I wanted to get the `SerializedProperty` sometimes, I can't say really but it just felt right to me to separate this concern into another private class. Then it didn't feel right to create a class every time because we are not really creating anything new but use something that exists, so that's why I made the constructor private and added the `GetBy` which also implies that there are multiple ways of getting that value. But I get your point, I just made the decision to do it this way, but it's really subjective.